### PR TITLE
CI: Add bump-version action to the release PR workflow

### PR DESCRIPTION
**Pull request by kminehart**
This doesn't change the behavior of the bunp-version action in a mentionable way.

It does add a version bump to the release PR workflow.

For the bump-version action I decided to use Dagger. I could use something like a new `make` target but I really want to be able to write and run the action locally and have it behave the same way remotely as it does when I run it locally. In my experience it's fantastic at that. Even for small things like `bump-version` it's nice to be able to run the whole thing in a docker container with the correct nodejs version.

### CODE REVIEW

**Comment by summerwollin** [comment]
Two questions: how has this been tested and how can we add more docs/comments for future engineers to more easily understand how to use?

**Review by summerwollin — APPROVED** [review]
Approving as seems to be a no-op but left a comment about testability and docs

File: pkg/build/actions/bump-version/main.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+	}
+
+	src := d.Host().Directory(filepath.Clean(*dir), dagger.HostDirectoryOpts{
+		Include: []string{
+			".nvmrc",
+			".yarn",
+			".yarnrc.yml",
+			"nx.json",
+			"project.json",
+			"lerna.json",
```
Comment (baldm0mma) [inline pkg/build/actions/bump-version/main.go:37]: aren't we killing `lerna` for `nx`?

**Comment by kminehart** [comment]
I tested this in a two ways:

1. You can run the `bump-version` and `release-pr` workflows with the `dry_run` option.
2. I ran it without the dry_run option to see what it would give me:
    * bump-version PR: https://github.com/grafana/grafana/pull/89377
    * release PR: https://github.com/grafana/grafana/pull/89488
    
To run it:

release-pr:
```
gh workflow run \
  --repo=grafana/grafana \
  --ref=km/bump-version\
  -f version=v11.0.0 \
  -f target=km/bump-version \
  -f backport=v10.1.x \
  -f dry_run=true \
  release-pr.yml
```

bump-version:

```
gh workflow run \
  bump-version.yml \
  --repo grafana/grafana \
  --ref km/bump-version \
  -f version=11.3.0-pre \
  -f dry_run=true
```

I'm really not sure what the answer is to documenting CI or github workflows yet to be honest...

**Review by baldm0mma — APPROVED** [review]
~~I have no idea how to test this, but the code looks good to me! 😁~~ I see how you tested this now. LGTM.

File: pkg/build/actions/bump-version/main.go (hunk 1 above)
Comment (kminehart) [reply pkg/build/actions/bump-version/main.go:37]: probably but i'm just copying what the existing behavior is. If anything I think we'll continue to have lerna call nx.

**Pull request merged**